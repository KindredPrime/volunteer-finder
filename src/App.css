@import url('https://fonts.googleapis.com/css2?family=Karla&family=Sura&family=Varela+Round&display=swap');

* {
  box-sizing: border-box;
}

body {
  background-color: #1A4161;
  color: white;
  margin-bottom: 20px;
}

h1 {
  text-shadow: -3px 0 black, 0 3px black, 3px 0 black, 0 -3px black;
  font-size: 34px;
  line-height: 48px;
}

h2 {
  text-shadow: -2px 0 black, 0 2px black, 2px 0 black, 0 -2px black;
  font-size: 26px;
  line-height: 36px;
}

p, legend {
  text-shadow: -1px 0 black, 0 1px black, 1px 0 black, 0 -1px black;
}

h1, h2, legend {
  font-family: 'Varela Round', sans-serif;
}

legend {
  font-size: 18px;
  line-height: 24px;
}

p, label {
  font-family: 'Karla', sans-serif;
}

p, label {
  font-size: 16px;
  line-height: 24px;
}

p {
  text-align: left;
}

section {
  border-top: 3px solid #747A82;
  margin: 20px 0px 20px 0px;
}

section > * {
  margin-left: 5%;
  margin-right: 5%;
}

button {
  background-color: #FF9100;
  color: black;
  cursor: pointer;
}

button:disabled {
  background-color: #747A82;
}

a {
  text-decoration: none;
  color: white;
}

a, button {
  font-family: 'Sura', serif;
}

form {
  padding: 10px;
}

fieldset, legend {
  margin-left: auto;
  margin-right: auto;
}

label {
  margin-right: 10px;
}

form button {
  margin-top: 10px;
}

fieldset {
  border: none;
}

legend {
  font-weight: bold;
}

p.error {
  color: #FF9100;
}

ul {
  padding: 0;
}

li {
  list-style: none;
}

.App {
  margin-top: 75px;
  text-align: center;
}

.App.expanded {
  margin-top: 138px;
}

@media screen and (min-width: 750px) {
  .App.expanded {
    margin-top: 75px;
  }
}

@media screen and (min-width: 1200px) {
  .App, .App.expanded {
    margin-top: 0px;
    margin-left: 17%;
  }
}