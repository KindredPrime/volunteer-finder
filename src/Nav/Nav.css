.Nav {
  background-color: black;
  padding: 15px;
  border-bottom: 1px solid #747A82;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  text-align: right;
}

.Nav *, .Nav__expanded * {
  color: white;
}

.Nav__expander {
  border: none;
  background: none;
  padding: 0;
  font-size: medium;
  cursor: pointer;
}

.Nav .links {
  display: none;
}

.Nav.expanded .links {
  display: flex;
  flex-direction: column;
  align-items: flex-end;
}

.Nav a {
  font-size: 16px;
}

.Nav.expanded a {
  text-decoration: none;
}

a.active {
  color: #FF9100;
}

@media screen and (min-width: 750px) {
  .Nav__expander {
    display: none;
  }

  .Nav .links, .Nav.expanded .links {
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
    align-items: center;
  }

  .Nav .links a, .Nav.expanded .links a {
    margin-left: 20px;
  }
}

@media screen and (min-width: 1200px) {
  .Nav {
    border-bottom: none;
    border-right: 1px solid #747A82;
    top: 0;
    right: 83%;
    bottom: 0;
    left: 0;
  }

  .Nav .links, .Nav.expanded .links {
    flex-direction: column;
    justify-content: flex-start;
    align-items: flex-end;
  }

  .Nav .links a, .Nav.expanded .links a {
    margin-left: 0px;
  }
}