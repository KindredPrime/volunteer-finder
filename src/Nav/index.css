.Nav {
  background-color: black;
  padding: 15px;
  border-bottom: 1px solid var(--border-color);
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  text-align: right;
}

.Nav * {
  color: white;
}

.Nav__expander {
  font-weight: normal;
  border: none;
  background: none;
  padding: 0;
  font-size: medium;
  cursor: pointer;
}

.Nav__expander:hover, .Nav__expander:focus {
  font-weight: bold;
  color: var(--accent-color);
}

/*
  Set the color for the FontAwesome icon
*/
.Nav__expander.expanded path {
  color: var(--accent-color);
}

/*
  For mobile devices, only display the nav links when the nav bar is expanded
*/
.Nav .links {
  display: none;
}

.Nav.expanded .links {
  display: flex;
  flex-direction: column;
  align-items: flex-end;
}


.Nav a {
  font-size: 16px;
}

.Nav.expanded a {
  text-decoration: none;
}

a.active {
  color: var(--accent-color);
}

/*
  Tablets
*/
@media screen and (min-width: 750px) {
  .Nav__expander {
    display: none;
  }

  /*
    For tablets and larger devices, there is no nav expander, so nav links are always "expanded"
  */
  .Nav .links, .Nav.expanded .links {
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
    align-items: center;
  }

  .Nav .links a, .Nav.expanded .links a {
    margin-left: 30px;
  }

  .Nav a {
    font-size: 18px;
  }
}

/*
  Desktop
*/
@media screen and (min-width: 1200px) {
  .Nav {
    border-bottom: none;
    border-right: 1px solid var(--border-color);
    top: 0;
    right: 85%;
    bottom: 0;
    left: 0;
    text-align: center;
    padding: 25px;
  }

  .Nav .links, .Nav.expanded .links {
    flex-direction: column;
    justify-content: flex-start;
  }

  .Nav .links a, .Nav.expanded .links a {
    margin-left: 0px;
    margin-bottom: 35px;
  }

  .Nav a {
    font-size: 20px;
  }
}