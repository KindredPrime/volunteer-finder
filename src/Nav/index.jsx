import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faBars } from '@fortawesome/free-solid-svg-icons';
import './index.css';

class Nav extends Component {
  render() {
    const { expanded, handleExpander } = this.props;

    return (
      <nav className={expanded ? "Nav expanded" : "Nav"} >
        <button
          className={expanded ? "Nav__expander expanded" : "Nav__expander"}
          type="button"
          onClick={() => handleExpander()}
        >
          <FontAwesomeIcon icon={faBars} />
        </button>

        <div className="links">
          <NavLink
            exact
            to="/"
            onClick={() => handleExpander()}
          >
            Home
          </NavLink>

          <NavLink
            to="/org-search"
            onClick={() => handleExpander()}
          >
            Search Organizations
          </NavLink>

          <NavLink
            to="/add-org"
            onClick={() => handleExpander()}
          >
            Add Organization
          </NavLink>
        </div>
      </nav>
    );
  }
}

export default Nav;