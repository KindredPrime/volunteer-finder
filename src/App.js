import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import { dummyOrgs, dummyCauses } from './dummyData';
import VolunteerContext from './VolunteerContext';
import Home from './Home/Home';
import Organization from './Organization/Organization';
import OrgSearch from './OrgSearch/OrgSearch';
import AddOrg from './AddOrg/AddOrg';
import './App.css';

class App extends Component {
  static contextType = VolunteerContext;

  state = {
    orgs: dummyOrgs,
    causes: dummyCauses
  };

  addOrg = (name, website, phone, email, address, description, causes) => {
    const { orgs } = this.state;
    const id = orgs.length + 1;
    const org = {
      id,
      name,
      website: website || 'Not Available',
      phone: phone || 'Not Available',
      email: email || 'Not Available',
      address: address || 'Not Available',
      description,
      causes
    };

    this.setState({
      orgs: [...orgs, org]
    });

    return id;
  }

  render() {
    const { orgs, causes } = this.state;
    const contextValue = {
      orgs,
      causes,
      addOrg: this.addOrg
    };

    return (
      <VolunteerContext.Provider value={contextValue}>
        <div className="App">
          <Route exact path="/" component={Home} />
          <Route path="/org/:id" component={Organization} />
          <Route path="/org-search" render={() => <OrgSearch pageLimit={10} />} />
          <Route path="/add-org" component={AddOrg} />
        </div>
      </VolunteerContext.Provider>
    );
  }
}

export default App;
