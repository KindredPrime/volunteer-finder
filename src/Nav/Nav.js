import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import VolunteerContext from '../VolunteerContext';
import './Nav.css';

class Nav extends Component {
  static contextType = VolunteerContext;

  state = {
    expanded: false
  };

  render() {
    const { expanded } = this.state;

    return (
      <nav className="Nav">
        <button 
          className="Nav__expander"
          type="button"
          onClick={() => this.setState({
            expanded: !this.state.expanded
          })}
        >
          Menu
        </button>
        {expanded && (
          <div className="Nav__expanded">
            <NavLink exact to="/">Home</NavLink>
            <NavLink to="/org-search">Search Organizations</NavLink>
            <NavLink to="/add-org">Add Organization</NavLink>
          </div>
        )}
      </nav>
    );
  }
}

export default Nav;