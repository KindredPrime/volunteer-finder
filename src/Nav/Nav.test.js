import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { render, screen } from '@testing-library/react';
import Nav from './Nav';
import App from '../App';

describe('Nav Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <Nav />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it(`expands and closes nav menu when 'Menu' is clicked`, () => {
    render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );

    userEvent.click(screen.getByRole('button', { name: 'Menu' }));
    expect(document.body).toMatchSnapshot();

    userEvent.click(screen.getByRole('button', { name: 'Menu' }));
    expect(document.body).toMatchSnapshot();
  });
});
