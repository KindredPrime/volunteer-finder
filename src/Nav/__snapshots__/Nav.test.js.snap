// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Component expands and closes nav menu when 'Menu' is clicked 1`] = `
<body>
  <div>
    <nav
      class="Nav"
    >
      <button
        class="Nav__expander"
        type="button"
      >
        Menu
      </button>
      <div
        class="Nav__expanded"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Home
        </a>
        <a
          href="/org-search"
        >
          Search Organizations
        </a>
        <a
          href="/add-org"
        >
          Add Organization
        </a>
      </div>
    </nav>
  </div>
</body>
`;

exports[`Nav Component expands and closes nav menu when 'Menu' is clicked 2`] = `
<body>
  <div>
    <nav
      class="Nav"
    >
      <button
        class="Nav__expander"
        type="button"
      >
        Menu
      </button>
    </nav>
  </div>
</body>
`;
